### ISSUE TITLE
Selecting Hidden Files in More Options Menu Opens Empty Window

### ISSUE THREAD

**Issue by thewizardstudio** [TeamAmaze/AmazeFileManager#956]
#### Expected Behavior
Choosing the Hidden Files option in the More Options menu brings up a window listing hidden files.

#### Actual Behavior
Choosing the Hidden Files option in the More Options menu brings up a window with no information in it.

#### How to Reproduce
1. Launch Amaze File Manager 3.2.1 on Android 8.1.0
2. Select the 3 Horizontal Lines Icon in the Upper Left
3. Toggle 'Show Hidden Files and Folders' to 'On'
4. Go to a Directory containing Hidden Files
5. Select the 3 Vertical Dots Icon in the Upper Right
6. Select 'Hidden Files'

---
    Device: Pixel 2 XL
    OS: Android 8.1.0
    App Version: 3.2.1
---

#### Recording of the Bug
![hiddenfilesdoesnothing](https://user-images.githubusercontent.com/35550916/35256639-ff8ab0de-ffb1-11e7-9dce-523d3b0eed7e.gif)

<br /><hr/><em>Posted on <a href="https://utopian.io/utopian-io/@thewizard.studio/selecting-hidden-files-in-more-options-menu-opens-empty-window">Utopian.io -  Rewarding Open Source Contributors</a></em><hr/>

**Comment by EmmanuelMess** [comment]
The actual problem here is a misleading naming of 'states'.

**Comment by bowiechen** [comment]
I agree, this is a very misleading phrasing. 

Fundamentally, we need to decide if we want this feature to be a purely cosmetic one (how Amaze treats files) or one that modifies the filesystem (how Amaze treats directories). Currently, Amaze hides the directories by adding `.nomedia` into the directories that user selected to hide. If we implement #1708, we will effectively cover the case where users want to hide this folder from other apps and free us to make the "hide file" feature to be consistent while treating files and directories.

**Comment by EmmanuelMess** [comment]
I think we should merge files hidden in the FS and files hidden in the app into a single _thing_: "files hidden". But, "hide" is not a tick box in the underlying file metadata. 

Hiding files in the FS usually means adding a dot to the name 'x' -> '.x'. This might seem a bit weird to some users (maybe a small text could explain?).

**Comment by bowiechen** [comment]
I think the "hide files" feature should **NOT** mutate the filesystem in any way. The reason behind this is because doing so creates more problem than it solves. 

> Hiding files in the FS usually means adding a dot to the name 'x' -> '.x'. 

In the case you raised, adding a leading `.` _does_ hide a directory/file from the rest of the filesystem, but that will break lots of references for other apps. For example, if the user hides `/storage/0/emulated/Android` and therefore changes the directory name to `.Android`, it breaks pretty much every app on their phone that uses the STORAGE permission. We should just "note" the file/directory should be hidden from view in our database. 

> All this is different to hiding directories from media explorers (by adding '.nomedia') and IMO should be kept as a separate 'thing' completely so as to not cause confusion while keeping this feature.

This is what #1708 is for. I think I phrased the feature well enough such that it isn't easily confused with hiding a file in Amaze. 

> We should move the show hidden files and folders tick box to the overflow menu in the appbar, removing the 'hidden files' list currently there.

Sure, that is one way to approach toggling the "hidden" status of a directory, but what about a file? I think we should move the "hide" option to the overflow menu of **each** file/directory. It's much more accessible.

**Comment by EmmanuelMess** [comment]
> I think the "hide files" feature should **NOT** mutate the filesystem in any way. The reason behind this is because doing so creates more problem than it solves.

I don't think we should limit out users from doing something just because it is dangerous. Maybe alert the user with a dialog, like when deleting a file.

> > We should move the show hidden files and folders tick box to the overflow menu in the appbar, removing the 'hidden files' list currently there.
> 
> Sure, that is one way to approach toggling the "hidden" status of a directory, but what about a file? I think we should move the "hide" option to the overflow menu of **each** file/directory. It's much more accessible.

I meant that the toggle to "show hidden files and directories" that's in the settings should be moved to the overflow menu of the appbar (not the context toolbar).

**Comment by bowiechen** [comment]
Right, we should not limit what a file manager should do just because it's dangerous. Hiding a folder from view as Amaze currently does with a file is consistent with that principle. 

What I am proposing is that we should stick with one method of hiding user files and not treat files and directories differently. Hiding files and directories through different methods breed confusion.

**Comment by EmmanuelMess** [comment]
> What I am proposing is that we should stick with one method (hiding from view and not mutate filesystem) of hiding user files and not treat files and directories differently. Hiding files and directories through different methods breed confusion.

I am sorry if I've created confusion but this is what I am proposing we have as features:

* Files **and** folders are hidden by adding a dot to their names "x" -> ".x".

* Folders are hidden from media explorers by adding a ".nomedia" file in them.

**Comment by SilverGreen93** [comment]
Can we have a checkbox in the menu for showing or hiding system hidden files instead? The current options is very misleading.

**Comment by Nikon6302** [comment]
Join (long walk to the settings to view a single dotfile).

As an option, you can separate 2 options for hiding the file and create a set-in that chooses the view of the file hiding. Or at the first start, ask the user how he wants to hide the files. (use the translator)
